SLE 2024-09 release notes (#221)
ni/install-systemlink-enterprise: release-notes/2024-09/README.md

# SystemLink Enterprise release 2024-09 Release Notes

The 2024-09 release for SystemLink Enterprise has been published to <https://downloads.artifacts.ni.com>. This update includes new features, bug fixes, and security updates. Work with your account representative to obtain credentials to access these artifacts. If you are not upgrading from the previous release, refer to past release notes to ensure you have addressed all required configuration changes.

## New Features and Behavior changes

- The Alarm Service automatically deletes active alarms that haven't been updated in a configurable number of days.
    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L469)
- The Alarm Service enforces configurable limits on the number of active alarms that can be created, as well as the total number of alarms that can be created.
    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L484)

> - The Schedule page has been added under Operations in the Navigation Pane. The users can now view all the scheduled Test Plans, across all Systems in the Lab, in a timeline view.

Reviewer: ```suggestion
- The Schedule page was added under Operations in the Navigation Pane. You can now view all the scheduled test plans, across all systems, in a timeline view.
```
```suggestion
- The Schedule page has been added under Operations in the Navigation Pane. The users can now view all the scheduled Test Plans, across all Systems in the Lab, in a timeline view.
```

- Privilege **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** was split to **List available notebooks** to list the published notebooks, and **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** to list and download notebooks for notebook development.

>     - This privilege is enabled by default for Collaborator, Data Maintainer and Systems Maintainer built-in roles.

Reviewer: ```suggestion
    - This privilege is enabled by default for Collaborator, Data Maintainer, and Systems Maintainer built-in roles.
```
```suggestion
    - This privilege is enabled by default for Collaborator, Data Maintainer and Systems Maintainer built-in roles.
```

    - For the Automated Agent, the **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** privilege will be removed, as the Automated Agent should not need to download the notebooks content.

> - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on Jupyter user servers to be used when calling SystemLink or other external APIs from notebooks.

Reviewer: ```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on servers that use Jupyter. These certificates are used when calling SystemLink or other external APIs from notebooks.
```
```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on Jupyter user servers to be used when calling SystemLink or other external APIs from notebooks.
```

> - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods, to be used when calling SystemLink or other external APIs from notebooks.

Reviewer: ```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods. These certificates are used when calling SystemLink or other external APIs from the pods.
```
```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods, to be used when calling SystemLink or other external APIs from notebooks.
```

| RabbitMQ Version | First SystemLink Enterprise Version | Last SystemLink Enterprise Version |
| ---------------- | ----------------------------------- | ---------------------------------- |
| 3.11.x           | 0.12.x                              | 0.15.x                             |
| 3.12.x           | 0.16.x                              | 0.24.x                             |
| 3.13.x           | 0.25.x                              | current                            |

## Bugs Fixed

Only customer facing bugs have been included in this list.

- [closed-bugs-sle-2024-09](https://github.com/ni/install-systemlink-enterprise/tree/2024-09/release-notes/2024-09/closed-bugs-sle-2024-09.xlsx)

## Software Bill of Materials and Notices

[SBOM](https://github.com/ni/install-systemlink-enterprise/tree/2024-09/release-notes/2024-09/sbom)

[Notices](https://github.com/ni/install-systemlink-enterprise/tree/2024-09/release-notes/2024-09/notices)

## Versions

> **Top Level Helm Chart:** `systemlink 0.29.49`

Reviewer 2: I'll need to bump these versions once the cherry-pick bugs are in.

Reviewer 2: I updated these with the latest.

    - Search for columns to plot from the axis selection grid.
    - Color traces by data table and by column metadata.
- Adjust how long the SystemLink Enterprise Alarm Service keeps your alarms - [Configuring Alarm Retention](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/configuring-alarm-retention.html).
    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L469)
- Adjust how many alarms you can create in the SystemLink Enterprise Alarm Service - [Configuring Alarm Limits](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/configuring-alarm-limits.html).
    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L484)
- Learn about the [performance metrics for the Dashboard Host Alarm Service](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/performance-metrics-dashboard-host-service.html).

> - Privilege **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** was split to **List available notebooks** to list the published notebooks, and **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** to list and download notebooks for notebook development.

Reviewer 3: ```suggestion
- The privilege **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** was split to **List available notebooks** and **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment**.
```

>     - This privilege is enabled by default for Collaborator, Data Maintainer, and Systems Maintainer built-in roles.

Reviewer 3: ```suggestion
    - These privileges are enabled by default for Collaborator, Data Maintainer, and Systems Maintainer built-in roles.
```

    - For the Automated Agent, the **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** privilege will be removed, as the Automated Agent should not need to download the notebooks content.
- Support for [Private Certificate Authorities](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/private-certificate-authorities.html).
    - Connect to MongoDB and S3 compatible interfaces that use certificates signed by a private authority.
    - Call SystemLink APIs from Jupyter Notebook when the SystemLink API endpoint is configured to use a certificate signed by a private authority.

>     - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on servers that use Jupyter. These certificates are used when calling SystemLink or other external APIs from notebooks.

Reviewer 3: ```suggestion
    - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
```

>     - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods. These certificates are used when calling SystemLink or other external APIs from the pods.

Reviewer 3: With changes suggested in the previous line we can remove this line because it will be worded to include both Jupyter user pods and notebook execution workers running Jupyter notebooks.

Author: Removed the line `- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods. These certificates are used when calling SystemLink or other external APIs from the pods.`

> - Added abilities to configure the rate-limit values for notebook APIs inside the Helm chart.

Reviewer 3: We need a reference to where and how this is done. This likely will require us to update the system link-value.yaml in this repo to expose this configuration. Alternatively we could reference the ni.com doc on this configuration should one exist.

Author: I did ask Hem Kumar Rao on Teams to check this input - for some reason I cannot tag his user: hemkumarrao-ni

Reviewer 4: This is just a rate-limiting implementation for some APIs(notebook APIs) in the WebApp service. This is very much similar to the rate-limiting done in the Notebook Execution Service mentioned just below in this document. This is not an enterprise helm level change but a service helm level change. 

Reference PR - [Pull Request 787094](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/787094): WebApp Service | Configure rate limit via Helm chart

We can rephrase it into 
`- WebApp Service
      - Implemented Rate limiting for the notebook APIs`
      
 @prestwick @CIakab-NI @MuthuRajan-S @JD-Robertson

Author: Thanks, @hemkumarrao-ni > I did replace the initial line with the text you had suggested:
- Implemented Rate limiting for the Notebook APIs, in the WebApp Service.

Reviewer 3: FYI I'm going to adjust the wording of this line. We also lack docs for this. I am not going to hold the release for that.  Bug https://dev.azure.com/ni/DevCentral/_workitems/edit/2605257 filed.

- Notebook Execution Service
    - Created a dedicated swagger page for Artifact APIs.
    - Implemented Rate limiting for the Artifact APIs.
    - Created S3 buckets for Artifact APIs.

> - Notebook Execution Service, Systems Management Service

Reviewer 3: ```suggestion
- Systems Management Service
```

    - Added support for deploying a custom CA root to all managed clients.

>         - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L119)

Reviewer 3: We should add a line here the mimic the structure and content included when discussion private CA support for `global.trustedCertificateSecrets` (line 22)

Author: I did rephrase this as below.
@stefanc18 Stefan Ciuprina, could you help review the 1st and 2nd line below - specifically to the 2nd line could you help suggest the right words. Thanks,

- Systems Management Service support for deploying a custom CA root to all managed clients.
    - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L119)

Reviewer 5: The wording looks good to me, but I see that it is mentioning custom certificates on Jupyter notebooks as a subsection of "Systems Management Service support for deploying a custom CA root to all managed clients", which are unrelated. This exact phrase seems to be already written on line 22.

Moreover, in the context of Systems Management Service, custom certificates deployed to managed clients is not done through "global.trustedCertificatesSecrets", but by setting `global.deployApiHostCertificateToSystems` to `true` and providing a single certificate secret at `global.apiHostCertificateSecret`.

Reviewer 3: I reworked all of these for clarity and accuracy

- Use the following new features when [visualizing data tables in data spaces](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/visualizing-data-tables-in-a-data-space.html).
    - Search for columns to plot from the axis selection grid.
    - Color traces by data table and by column metadata.

- Support for [Private Certificate Authorities](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/private-certificate-authorities.html).
    - Connect to MongoDB and S3 compatible interfaces that use certificates signed by a private authority.
        - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that connect to MongoDB and S3 compatible interfaces. These certificates are used when connecting to these external dependencies.
        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L113)
    - Call SystemLink APIs from Jupyter Notebook when the SystemLink API endpoint is configured to use a certificate signed by a private authority.

>         - Trusted certificates specified at `global.apiHostCertificateSecret` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.

Reviewer 5: `global.apiHostCertificateSecret` is used only for the certificate deployed from Systems Management to the systems, not for Jupyter notebooks. JupyterHub & Notebook Executions use the same `global.trustedCertificatesSecrets` that MongoDB and S3 use

Reviewer 3: Good catch. Updated.

>         - Trusted certificates specified at `global.trustedHostCertificateSecret` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.

Reviewer 5: ```suggestion
        - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
```

>         - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L119)

Reviewer 5: ```suggestion
        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L113)
```

Reviewer 3: fixed
